### PR TITLE
fast skip of none in pipeiterator

### PR DESCRIPTION
**Pull request by sh-rp**

### CODE REVIEW

File: dlt/extract/pipe.py [hunk 1]
```diff
@@ -732,7 +734,9 @@ def _get_source_item_round_robin(self) -> ResolvablePipeItem:
             # print(f"got {pipe.name} {pipe._pipe_id}")
             # register current pipe name during the execution of gen
             set_current_pipe_name(pipe.name)
-            item = next(gen)
+            item = None
+            while not item:
+                item = next(gen)
```
[the remark this answers is not in the document]
Comment (sh-rp) [reply dlt/extract/pipe.py]: @z3z1ma thanks for the feedback. I think I have changed it the way you suggest, and I have added a test case to see wether it works, but I am not sure those tests are right. if I comment out the new part of the code, the test will still pass and the simulated jobs are running parallel, which also totally makes sense from what I can read in the code. Can you give me a hint how to modify the test so it applies to your use case?

[the remark this answers is not in the document]
Comment (sh-rp) [reply dlt/extract/pipe.py]: ok ,sorry, now I get it, I think it's good now :)

File: dlt/extract/pipe.py [hunk 2]
```diff
@@ -718,21 +720,22 @@ def _get_source_item_current(self) -> ResolvablePipeItem:
...
-            gen, step, pipe, meta = self._sources[self._round_robin_index]
             # print(f"got {pipe.name} {pipe._pipe_id}")
             # register current pipe name during the execution of gen
-            set_current_pipe_name(pipe.name)
-            item = next(gen)
+            item = None
+            while item is None:
+                self._round_robin_index = (self._round_robin_index + 1) % sources_count
+                gen, step, pipe, meta = self._sources[self._round_robin_index]
+                set_current_pipe_name(pipe.name)
```
Comment (sh-rp) [inline dlt/extract/pipe.py:737]: can this go below the loop, or is this somehow used for exceptions that occur during the next call that follows?

Comment (rudolfix) [reply dlt/extract/pipe.py:737]: it should be set immediately when a new potential item is taken from _sources. so it looks good

**Comment by rudolfix** [comment]
@z3z1ma do you plan to review again?

**Comment by z3z1ma** [comment]
LGTM, my only question is why `self._initial_sources_count` is decremented in the round robin method but not in the current method.

**Review by z3z1ma — APPROVED** [review]

**Pull request merged**